server/game/Card.js

```diff
@@ -335,6 +335,7 @@ class Card extends PlayableObject {
                 {
                     when: {
                         onCardLeavesPlay: (event, context) =>
+                            event.card === context.source &&
```

Comment:
forgot to make sure the unit who called this was the one being destroyed, so ash spirits destroyed attackers no matter who died on their side. a little overpowered that way imo